server/msgset.go

```diff
@@ -113,7 +113,7 @@ func (a *Account) AddMsgSet(config *MsgSetConfig) (*MsgSet, error) {
 	}
 	// Setup the internal client.
 	c := s.createInternalJetStreamClient()
-	mset := &MsgSet{jsa: jsa, config: cfg, client: c, obs: make(map[string]*Observable)}
+	mset := &MsgSet{jsa: jsa, config: cfg, client: c, obs: make(map[string]*Observable), mu: sync.RWMutex{}}
```

Comment:
seems unlikely that this would be needed but I triggered this without it

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1439c7e]

goroutine 934 [running]:
github.com/nats-io/nats-server/v2/server.(*MsgSet).Name(0x0, 0x0, 0x0)
        /Users/rip/go/src/github.com/nats-io/nats-server/server/msgset.go:432 +0x2e
github.com/nats-io/nats-server/v2/server.(*Observable).sampleAck(0xc0000ab6c0, 0x1, 0x1)
        /Users/rip/go/src/github.com/nats-io/nats-server/server/observable.go:540 +0x68
github.com/nats-io/nats-server/v2/server.(*Observable).ackMsg(0xc0000ab6c0, 0x1, 0x1)
        /Users/rip/go/src/github.com/nats-io/nats-server/server/observable.go:561 +0x1a3
```

Reply:
Yea we should not need it, will fix elsewhere.

Reply:
msgset is nil.

Reply:
ah thats true, you're right it is the msgset, that really shouldnt happen :)